cmd/restic/cmd_rewrite.go

```diff
@@ -136,6 +136,10 @@ func rewriteSnapshot(ctx context.Context, repo *repository.Repository, sn *resti
 				if selectByName(path) {
 					return node
 				}
+				if sn.Summary != nil && node.Type == "file" {
```

Comment:
This will break if any directory is excluded.

Reply:
Instead please count all files returned by `return node` and use that to calculate the size from scratch.

Reply:
I'm not yet entirely sure that this is the best place to implement the size calculation. Here it is pretty hard to test it properly. What would be much more testable is SizeCalculationgRewriter that wraps `walker.NewTreeRewriter` and also the passed in `RewriteNode` function to count the remaining snapshot size.

To be a bit more specific I have something like the following in mind (I'm not entirely happy about the `func() SnapshotSize` part):
```go

type SnapshotSize struct {
	FileCount int
	FileSize  int64
}

func NewSizeCountingRewriter(rewriteNode NodeRewriteFunc) (*TreeRewriter, func() SnapshotSize) {
	count := 0
	size := int64(0)
	return NewTreeRewriter(RewriteOpts{
			RewriteNode: func(node *restic.Node, path string) *restic.Node {
				node = rewriteNode(node, path)
				if node != nil {
					// accounting
				}
				return nil
			},
			DisableNodeCache: true,
		}), func() SnapshotSize {
			return SnapshotSize{count, size}
		}
}
```

That would allow using the test infrastructure that already exists in `internal/walker/rewriter_test.go`. With that in place, a single test in `cmd_rewrite.go` would be enough to provide a decent test coverage.

Reply:
@MichaelEischer : Could you take another look? I ended up creating a function which accepts with snapshot summary as input to be updated in place. And added some tests too.